### PR TITLE
Refactors AI law announcing

### PR DESCRIPTION
**Pull request by SuperNovaa41**
## About The Pull Request
This had some weird list manipulation, and generally just a weird
structure. I changed it to hold every law it's announcing to be
contained within a list, rather than just have a list of Yeses and Nos.
The old functionality also increased the length of this list each time
laws were checked, meaning theoretically this could cause high memory
usage issues. Anyways... fixed all that.

## Why It's Good For The Game
Makes the code for ai law announcing better.

- [x] TODO: Fix MMIs

## Changelog

:cl:
refactor: Refactored AI law announcing.
/:cl:

### CODE REVIEW

File: code/datums/ai_laws.dm [hunk 1]
```diff
@@ -318,14 +318,20 @@
 /datum/ai_laws/proc/add_inherent_law(law)
 	if (!(law in inherent))
 		inherent += law
+		owner?.lawcheck += law
```
Comment (optimumtact) [inline code/datums/ai_laws.dm:321]: in what situation would owner ever be missing

Comment (optimumtact) [reply code/datums/ai_laws.dm:321]: same goes everywhere else

Comment (SuperNovaa41) [reply code/datums/ai_laws.dm:321]: I had it because it runtimed during testing.. however it seems to not happen anymore. I'll remove it

Comment (SuperNovaa41) [reply code/datums/ai_laws.dm]: nvm its because of MMIs

Comment (optimumtact) [reply code/datums/ai_laws.dm]: why because of MMI's?

this seems like a bug

Comment (SuperNovaa41) [reply code/datums/ai_laws.dm]: it is, ill fix it

Comment (SuperNovaa41) [reply code/datums/ai_laws.dm]: anyways, MMIs and empty AI cores get the ai_laws datum, so thats when owner is null

**Comment by SuperNovaa41** [comment]
Ok works now, fixed the whole null owner business

**Pull request merged**